src/main/java/org/jenkinsci/plugins/workflow/support/steps/build/DownstreamBuildAction.java

```diff
@@ -0,0 +1,53 @@
+package org.jenkinsci.plugins.workflow.support.steps.build;
+
+import edu.umd.cs.findbugs.annotations.CheckForNull;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.model.InvisibleAction;
+import hudson.model.Item;
+import hudson.model.ItemGroup;
+import hudson.model.Run;
+import org.springframework.security.access.AccessDeniedException;
+
+/**
+ * Tracks the downstream build triggered by the {@code build} step.
+ * <p>Note that {@link #getBuildNumber} may return null if the build is in the queue or gets cancelled from the queue.
+ * <p>Keep in mind that {@link #getBuild} may need to load the build, and so should be used carefully.
+ *
+ * @see BuildUpstreamCause
+ * @see BuildUpstreamNodeAction
+ */
+public class DownstreamBuildAction extends InvisibleAction {
+    private final String jobFullName;
+    private Integer buildNumber;
+    private String buildId;
```

Comment:
This is simply `${jobFullName}#${buildNumber}` so it does not make sense to store separately.

Reply:
Yeah, I prefer not to try to interpret/parse otherwise opaque IDs outside of the context where they are defined, but I guess it is already effectively impossible to change the structure of `Run.getExternalizableId` so we can just build the ID ourselves from the components.

Reply:
Yes I would consider the format of the externalizable id to effectively be a committed API. There are certainly other pieces of code which assume it can be constructed using string concatenation.

Reply:
https://github.com/jenkinsci/pipeline-build-step-plugin/pull/127/commits/1a96a2b25711b9f2b6cde3632747747e67faaaf4